src/packaging/version.py

```diff
@@ -450,7 +450,7 @@ def micro(self) -> int:
 
 
 def _parse_letter_version(
-    letter: str, number: Union[str, bytes, SupportsInt]
+    letter: Optional[str], number: Union[str, bytes, SupportsInt, None]
```

Comment:
I can see why the `number` change, but I don't' see any use of `None` for `letter` in this method. Can you explain why you are suggesting that change?

Reply:
Looks like external code such as `pkg_resources` is passing in the `None`.